lib/topological_inventory/openshift/collectors_pool.rb

```diff
@@ -0,0 +1,85 @@
+require "config"
+require "topological_inventory/openshift/collector"
+require "topological_inventory/openshift/logging"
+
+module TopologicalInventory::Openshift
+  class CollectorsPool
+    include Logging
+
+    def initialize(config_name, metrics, poll_time: 10)
+      self.collectors  = {}
+      self.collector_threads = {}
+      self.config_name = config_name
+      self.metrics     = metrics
+      self.poll_time   = poll_time
+    end
+
+    def run!
+      loop do
+        reload_config
+
+        remove_old_collectors
+        add_new_collectors
+
+        sleep(poll_time)
+      end
+    end
+
+    def stop!
+      collectors.each_value { |collector| collector.stop }
```

Comment:
Should be able to do `.each_value(&:stop)`

Reply:
is it neccessary? I liked ruby syntax because it was very simple to read...this is not as nice (I don't like it :)

Reply:
:man_shrugging: it is personal preference but as a project we have adopted SymbolProc (https://www.rubydoc.info/gems/rubocop/0.69.0/RuboCop/Cop/Style/SymbolProc) so for consistency with the rest of the project I prefer it.  I don't really have a preference specifically to either one but the rest of the project uses the latter